src/Elmish/Enzyme.purs

```diff
@@ -211,38 +281,33 @@ count selector = E.count selector =<< ask
 -- | re-render. For some reason, Enzyme won't pick up the changes automatically.
 -- | See https://enzymejs.github.io/enzyme/docs/api/ReactWrapper/update.html for
 -- | more info.
+-- |
+-- | NOTE: this only works on the "root" element, which means it cannot be
+-- | called inside `withSelector` or `withElement`.
 update :: EnzymeM Unit
-update =
-  E.update =<< ask
+update = E.update =<< ask
```

Comment:
👍🏻  We could probably enforce this in the types, too. Maybe `EnzymeM` could have an extra parameter indicating whether it’s the root.